detection_rules/custom_rules.py

```diff
@@ -7,43 +7,120 @@
 from pathlib import Path
 
 import click
+import yaml
 
 from .main import root
-from .utils import get_etc_path
+from .utils import get_etc_path, load_etc_dump, ROOT_DIR
 
+from semver import Version
 
 DEFAULT_CONFIG_PATH = Path(get_etc_path('_config.yaml'))
+CUSTOM_RULES_DOC_PATH = Path(ROOT_DIR).joinpath('docs', 'custom-rules.md')
 
 
 @root.group('custom-rules')
 def custom_rules():
     """Commands for supporting custom rules."""
 
 
-@custom_rules.command('init-config')
+def create_config_content() -> str:
+    """Create the content for the _config.yaml file."""
+    # Base structure of the configuration
+    config_content = {
+        'rule_dirs': ['rules', 'rules_building_block'],
+        'files': {
+            'deprecated_rules': 'etc/deprecated_rules.json',
+            'packages': 'etc/packages.yml',
+            'stack_schema_map': 'etc/stack-schema-map.yaml',
+            'version_lock': 'etc/version.lock.json',
+        },
+        'testing': {
+            'config': 'etc/test_config.yaml'
+        }
+    }
+
+    return yaml.safe_dump(config_content, default_flow_style=False)
+
+
+def create_test_config_content() -> str:
+    """Generate the content for the test_config.yaml with special content and references."""
+    example_test_config_path = DEFAULT_CONFIG_PATH.parent.joinpath("example_test_config.yaml")
+    content = f"# For more details, refer to the example configuration:\n# {example_test_config_path}\n" \
+              "# Define tests to explicitly bypass, with all others being run.\n" \
+              "# To run all tests, set bypass to empty or leave this file commented out.\n\n" \
+              "unit_tests:\n  bypass:\n#  - tests.test_all_rules.TestValidRules.test_schema_and_dupes\n" \
+              "#  - tests.test_packages.TestRegistryPackage.test_registry_package_config\n"
+
+    return content
+
+
+@custom_rules.command('setup-config')
 @click.argument('directory', type=Path)
-def init_config(directory: Path):
-    """Initialize the custom rules configuration."""
+@click.argument('kibana-version', type=str, default=load_etc_dump('packages.yml')['package']['name'])
+@click.option('--delete-config', is_flag=True, help="Delete the existing _config.yaml file.")
+def setup_config(directory: Path, kibana_version: str, delete_config: bool):
+    """Setup the custom rules configuration directory and files with defaults."""
+
+    if delete_config:
+        config = directory / '_config.yaml'
+        if config.exists():
+            config.unlink()
+            click.echo(f'Deleted existing _config.yaml file: {config}')
+
```

Comment:
the way this is implemented is more of an _overwrite_ than _delete_, since the file is going to be written anyway. Thoughts on this?

```suggestion
    config = directory / '_config.yaml'
    if not overwrite and config.exists():
        raise FileExistsError(f'{config} already exists. Use --overwrite to update')
```

(then you can remove config from L71)